src/libgit2/refs.c

```diff
@@ -819,7 +819,7 @@ int git_reference_list(
 
 static int is_valid_ref_char(char ch)
 {
-	if ((unsigned) ch <= ' ')
+	if ((unsigned) ch <= ' ' || ch == '\177') /* ASCII control characters */
```

Comment:
This check is still incomplete as it would accept ASCII chars between ' ' (0x32) and '(' (0x40) such as `#` or `'`, right?

Reply:
Argh, sorry: `\040` = 0x32, too tired